repositories.md

```diff
@@ -5,9 +5,75 @@ css_id: repositories
+
+* The Time Server will expose a public interface allowing primaries to communicate with it. This communication MAY occur over FTP, FTPS, SFTP, HTTP, HTTPS, or another transport control of the implementor's choice.
+
+* Rotation of the The Time Server's key is performed by listing the new key in the Director's Root metadata, in the same manner as other role keys are listed, and also in the custom field of the Director Repository's Targets metadata (for partial verification secondaries).
+
+#### Changes to the Director repository
+If a Time Server is in use, a representation of the Time Server public key is CONDITIONALLY REQUIRED in Director repository root metadata.
+
+If a Time Server is implemented AND partial-verification secondaries are used, the following metadata is CONDITIONALLY REQUIRED in the Director repository's Targets metadata:
+
+* A representation of the public key(s) for the Time Server, similar to the representation of public keys in Root metadata.
+
+Listing the public key of the Time Server in Director targets metadata is necessary to allow partial-verification secondaries to perform time server key rotation.
+
+#### Changes to a Primary
+
+If the Time Server is implemented, the primary is CONDITIONALLY REQUIRED to use the following procedure to verify the time. This procedure occurs after the vehicle version manifest is sent and will fulfill the ["Download and check current time"](https://uptane.github.io/papers/ieee-isto-6100.1.0.0.uptane-standard.html#check_time_primary) step of the Uptane Standard.
+
+1. Gather the tokens from each secondary ECU's version report.
+2. Send the list of tokens to the Time Server to fetch the current time. The Time Server responds as described in [Time Server](#time_server), providing a cryptographic attestation of the last known time.
+3. If the Time Server's response meets the criteria below, update the primary ECU's clock and retain the Time Server's response for distribution to secondary ECUs, otherwise discard it and proceed without an updated time.  The criteria for checking the Time Server's response are:
+  - The signature over the Time Server's response is valid.
+  - All the tokens provided to the Time Server have been included in the response.
+  - The time in the Time Server's response is later than the last time verified in this manner.
+
+#### ECU Version Report
+
+The ECU version report from each Secondary will contain a token to be sent to the Time Server. This token SHOULD be unique for each update cycle to prevent a replay. As updates are relatively infrequent and there are a large number of possible tokens, the token will be able to be unique for every update.
```

Comment:
```suggestion
The ECU version report from each Secondary will contain a token to be sent to the Time Server. This token SHOULD be unique for each update cycle to prevent a replay. Since we expect updates to be relatively infrequent (e.g., due to limited number of write cycles), and that there are a large number of possible tokens, it should be possible to produce a unique token for every update.
```